src/models.py

```diff
@@ -1,19 +1,238 @@
+            if social is None:
+                return None
+            else:
+                return social.serialize()
+    
+    def update_social(new_user_id, new_id_social):
+        user = User.get_user(new_user_id)
+        if user is None:
+            return User.get_user(new_user_id)
+        else:
+          social = Social.query.filter_by(user_id=new_user_id,id=new_id_social, is_active=True).first()
+          if social is None:
+              return None
+          else:
+              if "social_name" in body:
+                  social.social_name = body["social_name"]
+              if "email" in body:
+                  social.email = body["email"]
+              if "username" in body:
+                  social.username = body["username"]
+              if "password" in body:
+                  social.password = body["password"]
+              db.session.commit()
+              return "social media account with id: " + str(social.id) + " updated"
+
+class Post(db.Model):
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    date = db.Column(db.Date, nullable=False, unique=False)
+    description = db.Column(db.Text, nullable=False, unique=False)
+    multimedias = db.relationship('Multimedia', backref='post', lazy=True)
+    id_social = db.Column(db.Integer, db.ForeignKey('social.id'))
+    is_active = db.Column(db.Boolean, default=True)
+        
+    def __repr__(self):
+        return f"Post {self.description}"
+
+    def serialize(self):
+        return {
+            "id": self.id,
+            "date": self.date,
+            "id_social": self.id_social
+        }
+    
+    def get_post(new_id_user,new_id_social,new_id_post):
+        social = Social.get_social(new_id_user,new_id_social)
+        user = User.get_user(new_id_user)
+        if user is None or social is None:
+            return None
+        else:
+            post = Post.query.filter_by(id_social=new_id_social,id=new_id_post, is_active=True).first()
+            if post is None:
+                return None
+            else:
+                return post.serialize()
+    
+    def get_all_post(new_id_user,new_id_social):
+        social = Social.get_social(new_id_user,new_id_social)
+        user = User.get_user(new_id_user)
+        if user is None or social is None:
+            return None
+        else:
+            posts = Post.query.filter_by(id_social=new_id_social, is_active=True)
+            posts = list(map(lambda x: x.serialize(), posts))
+            if len(posts) is 0:
+                return None
+            else:
+                return posts
+            return posts
+    
+    def update_post(new_id_user,new_id_social,new_id_post,body):
+        social= Social.get_social(new_id_user,new_id_social)
+        post = Post.query.filter_by(id_social=new_id_social,id=new_id_post, is_active=True).first()
+        if post is None or social is None:
```

Comment:
```suggestion
        post = Post.query.filter_by(id_social=new_id_social,id=new_id_post, is_active=True).first()
        if not post:
```
Igual que arriba, no es necesario comprobar social si ya está el id_social en el filtro del post.